nemo/collections/nlp/callbacks/sgd_callback.py

```diff
@@ -157,13 +157,14 @@ def eval_epochs_done_callback(
     prediction_dir,
     state_tracker,
     eval_debug,
+    dialogues_processor,
     schema_emb_preprocessor,
     joint_acc_across_turn,
     no_fuzzy_match,
 ):
     # added for debugging
     in_domain_services = get_in_domain_services(
-        os.path.join(data_dir, eval_dataset, "schema.json"), os.path.join(data_dir, "train", "schema.json")
+        os.path.join(data_dir, eval_dataset, "schema.json"), dialogues_processor._seen_services["train"]
     )
```

Comment:
maybe add a get_seen_services(dataset_split) method to the dialogues_processor class instead of using _seen_services?